dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/AccessLogDataFormatter.java

```diff
@@ -20,25 +20,36 @@
 public class AccessLogDataFormatter {
 
     // list of actions to populate JSONObjectBuilder
-    private final ArrayList<JsonFieldAdder> jsonFieldAdders = new ArrayList<JsonFieldAdder>();
+    private ArrayList<JsonFieldAdder> jsonFieldAdders = new ArrayList<JsonFieldAdder>();
 
-    public AccessLogDataFormatter() {
+    public static class AccessLogDataFormatterBuilder {
+        private final ArrayList<JsonFieldAdder> jsonFieldAdders = new ArrayList<JsonFieldAdder>();
+
+        public AccessLogDataFormatterBuilder() {
+        }
+
+        public AccessLogDataFormatterBuilder add(JsonFieldAdder jsonFieldAdder) {
+            this.jsonFieldAdders.add(jsonFieldAdder);
+            return this;
+        }
+
+        public AccessLogDataFormatter build() {
+            AccessLogDataFormatter formatter = new AccessLogDataFormatter(this);
+            return formatter;
+        }
 
     }
 
-    // adds another JsonFieldAdder to the list
-    public AccessLogDataFormatter add(JsonFieldAdder jsonFieldAdder) {
-        this.jsonFieldAdders.add(jsonFieldAdder);
-        return this;
+    public AccessLogDataFormatter(AccessLogDataFormatterBuilder builder) {
```

Comment:
make private

Reply:
If it's private then AccessLogSource can't access it since it's in http transport. It needs to be public, or else we'd have to make our list of formatters into a list of formatterBuilders, I think

Reply:
I think you should be able to call `AccessLogDataFormatter aldf = accessLogDataFormatterBuilder.build();` , no?

Reply:
yep, you're right, I don't know why I wasn't using that to begin with!